operations.ts

```diff
@@ -0,0 +1,138 @@
+import { Type, UNKNOWN } from './value';
+
+/**
+ * The expected return type of a typical binary math expression, given
+ * its input types.
+ *
+ * @param a The type of the first argument.
+ * @param b The type of the second argument.
+ *
+ * @returns The expected type of the operation.
+ */
+export function binaryMathReturnType(a: Type, b: Type): Type | typeof UNKNOWN {
+  // TODO: This is an extremely error-prone approach to this problem.
```

Comment:
The result type here is a little muddied. Right now it either:

- returns a Type, when known
- returns UNKNOWN when unknown
- would probably throw if the operation would fail

I said in a prior comment that `UNKNOWN` should probably be `Type.Unknown`. I also think that, in this context, `Type.Any` actually makes more sense. But, like the case with `wouldCast`/`wouldConvert`, I probably don't want to throw an Error. The difference is between:

```ts

try {
  const b = this.stack.pop();
  const a = this.stack.pop();
  const rt = binaryMathReturnType(a, b);
  this.stack.push(rt);
} catch (err) {
  // 1. Error gets thrown out
  // 2. Must synchronize to the next line
  this.syntaxError(cmd, 'Operation is invalid');
}
```

vs:

```ts
const b = this.stack.pop();
const a = this.stack.pop();
const rt = binaryMathReturnType(a, b);
if (rt === Type.Failure) {
  this.typeError(cmd, '+', a, b);
}
// Instead of synchronizing, we can keep the `Type.Failure` in the stack and show errors for cascading failures
this.stack.push(rt);
```